tools/IoTEdgeDevOps/VstsPipelineSync/DictionaryExtension.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace VstsPipelineSync
+{
+    using System.Collections.Generic;
+
+    public static class DictionaryExtension
+    {
+        public static TValue GetIfExists<TKey, TValue>(this Dictionary<TKey, TValue> dict, TKey key)
+        {
+            if (dict.ContainsKey(key))
+            {
+                return dict[key];
+            }
+
+            return default(TValue);
+        }
+
+        public static bool Upsert<TKey, TValue>(this Dictionary<TKey, TValue> dict, TKey key, TValue value)
```

Comment:
Did you mean to do this? 
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.tryadd?view=netcore-3.1

Note: please read the "Remark" section

Reply:
From the remarks:
"TryAdd doesn't override the element if the element with the given key exists in the dictionary"

So this is doing something different.